lib/rules/yoda.js

```diff
@@ -245,11 +234,23 @@ module.exports = {
             function isBetweenTest() {
                 let leftLiteral, rightLiteral;
 
-                return (node.operator === "&&" &&
-                    (leftLiteral = getNormalizedLiteral(left.left)) &&
-                    (rightLiteral = getNormalizedLiteral(right.right, Number.POSITIVE_INFINITY)) &&
-                    leftLiteral.value <= rightLiteral.value &&
-                    same(left.right, right.left));
+                if (node.operator === "&&" && same(left.right, right.left)) {
+                    leftLiteral = getNormalizedLiteral(left.left);
+                    rightLiteral = getNormalizedLiteral(right.right);
+
+                    if (leftLiteral === null && rightLiteral === null) {
+                        return false;
+                    }
+
+                    if (leftLiteral !== null && rightLiteral === null || rightLiteral !== null && leftLiteral === null) {
```

Comment:
With this change, the rule will now allow:

```js
/* eslint yoda: ["error", "always", { exceptRange: true }] */

if (x < 1 && 1 < 2) {}
```

which looks good, given that the actual master version already allows this:

```js
/* eslint yoda: ["error", "never", { exceptRange: true }] */

if (1 < 2 && 2 < x) {}
```

I'm not sure if it was the original intention to allow the second example, but the behavior will be at least consistent after this PR.

Reply:
But as per `exceptRange` is considered, it should be lint free right . !

Reply:
Nothing to do. I just wanted to leave a note that this PR also changes the behavior of this rule in the first example (which actually makes it consistent with the second example).

Reply:
ohh ok cool 👍